governor/tests/keyed_dashmap.rs

```diff
@@ -46,6 +46,215 @@ fn rejects_too_many() {
     }
 }
 
+#[test]
+fn rejects_too_many_n() {
+    let clock = FakeRelativeClock::default();
+    let lb = RateLimiter::dashmap_with_clock(Quota::per_second(nonzero!(2u32)), &clock);
+    let ms = Duration::from_millis(1);
+
+    for key in KEYS {
+        // use up our burst capacity (2 in the first second):
+        assert!(
+            lb.check_key_n(key, nonzero!(2u32)).is_ok(),
+            "Now: {:?}",
+            clock.now()
+        );
```

Comment:
codecov warns that lines in this multi-line assert aren't covered which... yeah ):

I think the easiest way to get rid of that is to reformat the condition you're checking in a let-bound variable and then assert on the variable.
```suggestion
        let res = lb.check_key_n(key, nonzero!(2u32));
        assert!(res.is_ok(), "Now: {:?}", clock.now());
```

(would be neat if we had `assert_matches` for these checks, but that's not stable yet)

Reply:
I think I got a solution. What is actually not covered here is the case the `check_n_key` fails.